libraries/AP_Scripting/examples/ahrs-set-origin.lua

```diff
@@ -0,0 +1,22 @@
+-- example script for using "set_origin()"
+-- sets the ekf origin if not already set
+
+
+function update ()
+
+    home = ahrs:get_home()
+    origin = ahrs:get_origin()
+    location = Location() location:lat(-353632640) location:lng(1491652352) location:alt(58409)
+
+    if origin then
+        gcs:send_text(0, "Unable to set EKF origin: already set")
+    return
+    else
+        ahrs:set_origin(location)
```

Comment:
you should check the return of this, make sure it has actually set.

Reply:
^^This: I still had gps enabled and it prints "Set" a bunch of times until it gets a lock and the origin is finally set.

Reply:
If you did check it you could do something like this and bring back your printing
```suggestion
    if ahrs:set_origin(location) then
        gcs:send_text(0, string.format("Origin Set - Lat:%.7f Long:%.7f Alt:%.1f", location:lat()/10000000, location:lng()/10000000, location:alt()/100))
    else
        gcs:send_text(0, string.format("Origin Refused to be Set"))
    end
```